src/coprocessor/dag/dag.rs

```diff
@@ -48,7 +48,19 @@ impl<S: Snapshot + 'static> DAGContext<S> {
         req_ctx: ReqContext,
         batch_row_limit: usize,
     ) -> Result<Self> {
-        let mut eval_cfg = box_try!(EvalConfig::new(req.get_time_zone_offset(), req.get_flags(),));
+        let mut eval_cfg = {
+            let time_zone_name = if req.has_time_zone_name() {
```

Comment:
Could we make `time_zone_name` to `None` as default value and set it when existing just like `set_max_warning_cnt` below, so we needn't update so many cases in test(seems they do not care about it).

Reply:
Nice idea. BTW I think we can refine it to a XxxBuilder style, which can be handed over to contributors.

Reply:
I found out that in fact currently we specify "time_zone_offset", which is not what we cared in tests in most cases as well 😢